src/OSPSuite.R/Services/ConcurrentSimulationRunner.cs

```diff
@@ -98,22 +98,23 @@ public void Clear()
 
       private Task initializeBatches()
       {
-         return _concurrencyManager.RunAsync
+         //The batch creation is expensive so we store the created batches from one RunConcurrently call
+         //to the next one. It might happen though that the later call needs more batches than the former
+         //so for each needed batch, we create a new one.
+         //The iteration occurs on the list of _listOfSettingsForConcurrentRunSimulationBatch (over different
+         //simulation objects), taking for each settings (or simulation) a list with the missing batches 
+         //(one for each MissingBatchesCount) to add a new batch on such a settings object
+         var batches = _listOfConcurrentRunSimulationBatch.SelectMany
          (
-            //The batch creation is expensive so we store the created batches from one RunConcurrently call
-            //to the next one. It might happen though that the later call needs more batches than the former
-            //so for each needed batch, we create a new one.
-            //The iteration occurs on the list of _listOfSettingsForConcurrentRunSimulationBatch (over different
-            //simulation objects), taking for each settings (or simulation) a list with the missing batches 
-            //(one for each MissingBatchesCount) to add a new batch on such a settings object
-            _listOfConcurrentRunSimulationBatch.SelectMany
-            (
-               settings => Enumerable.Range(0, settings.MissingBatchesCount).Select(_ => settings)
-            ).ToList(),
-            (settings, ct) => settings.AddNewBatch(),
+            batch => Enumerable.Range(0, batch.MissingBatchesCount).Select(_ => batch)
+         ).ToList();
+
+         return _concurrencyManager.RunAsync(
+            batches,
+            //Use {} to specify using the overload without return value
+            (batch, ct) => { batch.AddNewBatch(); },
```

Comment:
This is the only downside of using the same name... I need to use the {} to that it picks the right overload